src/main/scala/org/citegraph/analytics/Analytics.scala

```diff
@@ -29,25 +24,18 @@ class Analytics(sparkSession: SparkSession, citationsDF: DataFrame, publishedDat
       |1994| 11568|
       |... |...   |
       +----+------+
-     */
-
+   */
+  def getEdgeCountsByYear(df: DataFrame): DataFrame = {
     publishedDatesDF
-      .join(df, publishedDatesDF("id") === df("from"))
-      .select("year", "count")
-      .groupBy("year")
-      .sum("count")
-      .sort(col("year"))
-      //      .withColumnRenamed("year", "edgeYear")
-      .withColumnRenamed("sum(count)", "e(t)")
+      .join(df, publishedDatesDF("id") === df("from"))  // Inner join to publishedDates on the paper id
+      .select("year", "count")  // Select only the "year" and "count" columns
+      .groupBy("year").sum("count")  // Group by "year" and sum up the counts
+      .sort(col("year"))  // Sort by "year"
+      .withColumnRenamed("sum(count)", "e(t)")  // Rename "sum(count)" column to "e(t)"
```

Comment:
These types of comments are good. Helps you remember 6 months from now what you were thinking when you wrote the code. Thanks for adding them.